packages/api/src/external/hie/tally-doc-query-progress-with-source.ts

```diff
@@ -0,0 +1,119 @@
+import { DocumentQueryProgress, Progress } from "@metriport/core/domain/document-query";
+import { getStatusFromProgress } from "@metriport/core/domain/document-query";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { PatientExternalData } from "@metriport/core/domain//patient";
+import { ProgressType } from "@metriport/core/domain/document-query";
+import { Patient } from "@metriport/core/domain/patient";
+import { PatientModel } from "../../models/medical/patient";
+import { executeOnDBTx } from "../../models/transaction-wrapper";
+import { getPatientOrFail } from "../../command/medical/patient/get-patient";
+import { handleWebhookBeingSent } from "../../command/medical/document/process-doc-query-webhook";
+import { aggregateAndSetHIEProgresses, HIEPatientData } from "./set-doc-query-progress-with-source";
+
+type DynamicProgress = Pick<Progress, "successful" | "errors">;
+
+export type TallyDocQueryProgressWithSource = {
+  source: MedicalDataSource;
+  patient: Pick<Patient, "id" | "cxId">;
+  type: ProgressType;
+  progress: DynamicProgress;
+  requestId: string;
+};
+
+/**
+ * Updates the successful and error count for the given HIE which is then aggregated
+ * to the patient's document query progress. setDocQueryProgressWithSource to update
+ * the total and status.
+ *
+ * @returns
+ */
+export async function tallyDocQueryProgressWithSource({
```

Comment:
nit: I feel like the function name and docstring obfuscate a little that this is now a function that also triggers the sending of the webhooks via downstream call of `handleWebhooksBeingSent`

Reply:
Ill update the jsdoc to be more clear